app/install/source.md

```diff
@@ -11,90 +11,166 @@ redirect_from: /install/compile/
-    Kong being an OpenResty application, you must follow the OpenResty
-    [installation instructions](https://openresty.org/en/installation.html).
-    You will need [OpenSSL](https://www.openssl.org/) and
-    [PCRE](http://www.pcre.org/) to compile OpenResty, and to at least use the
-    following compilation options:
+**About the dependencies**
+
+Kong depends on several components (the versions mentioned are the preferred ones):
+- [OpenResty {{openresty_version}}](https://openresty.org/en/installation.html).
+- [Nginx Patches](https://github.com/Kong/kong-build-tools/tree/master/openresty-build-tools/patches)
+- [OpenResty Patches](https://github.com/Kong/kong-build-tools/tree/master/openresty-patches/patches/{{openresty_version}})
+- [Kong Nginx Module](https://github.com/Kong/lua-kong-nginx-module/)
+- [OpenSSL {{openssl_version}}](https://www.openssl.org/)
+- [PCRE {{pcre_version}}](http://www.pcre.org/)
+- [LuaRocks {{luarocks_version}}](https://github.com/luarocks/luarocks/wiki/Download)
+- [LibYAML {{libyaml_version}}](https://pyyaml.org/wiki/LibYAML)
+- [go-pluginserver {{gopluginserver_version}}](https://github.com/Kong/go-pluginserver)
+    
+It is not trivial to get everything built correctly, thus we provide a
+[kong-build-tools](https://github.com/Kong/kong-build-tools/).
+    
+1. **Build the dependencies** 
+
+   At first, you need to clone the `kong-build-tools`:
+   
+   ```bash
+   $ mkdir kong-sources
+   $ cd kong-sources
+   $ git clone git@github.com:Kong/kong-build-tools.git
+   ```
+   
+   Then you need to build the dependencies:
+   
+   ```bash
+   $ ./kong-build-tools/openresty-build-tools/kong-ngx-build \
+       --prefix deps \
+       --work work \
+       --openresty {{openresty_version}} \
+       --openssl {{openssl_version}} \
+       --kong-nginx-module master \
+       --luarocks {{luarocks_version}} \
+       --pcre {{pcre_version}} \
+       --jobs 6 \
+       --force
+   ```
+   
+   Now, you should have the dependencies built, but before we continue,
+   let's export some environment variables:
+   
+   ```bash
+   $ export OPENSSL_DIR=$(pwd)/deps/openssl
+   $ export PATH=$(pwd)/deps/openresty/bin:$PATH
+   $ export PATH=$(pwd)/deps/openresty/nginx/sbin:$PATH
+   $ export PATH=$(pwd)/deps/openssl/bin:$PATH
+   $ export PATH=$(pwd)/deps/luarocks/bin:$PATH
+   ```
+   
+   Check everything is working:
+   
+   ```bash
+   $ nginx -V
+   $ resty -v
+   $ openresty -V
+   $ openssl version -a
+   $ luarocks --version
+   ```
+   
+   You will also need [`LibYAML {{libyaml_version}}`](https://pyyaml.org/wiki/LibYAML)
+   that you can install either manually or by using your operating system packages:
+   - Ubuntu: `apt install libyaml-dev`
+   - Fedora: `dnf install libyaml-devel`
+   - macOS: `brew install libyaml`
+   
+   At this point you may continue to the next step, but if you are interested
+   in running Go plugins, please follow.
```

Comment:
Suggestion: let's link to the Go guide instead of writing explicit instructions over here, as the guide contains more info. Something like "At this point you may continue to the next step, but if you are interested in running Go plugins, please follow the [Go guide][go]."

Then we need to add the link at the end
```
[go]: /{{page.kong_version}}/go
```